src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,218 @@
+import React, { useState, Dispatch, SetStateAction } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import {
+	CREATE_COURSE,
+	TITLE,
+	DESCRIPTION,
+	ADD_AUTHOR,
+	DELETE_AUTHOR,
+	AUTHORS,
+	DURATION,
+	COURSE_AUTHORS,
+	EMPTY_AUTHORS,
+	TITLE_PLACEHOLDER,
+	DESCR_PLACEHOLDER,
+	DURATION_PLACEHOLDER,
+	CREATE_AUTHOR,
+	AUTHOR_NAME,
+	AUTHOR_PLACEHOLDER,
+	COURSE_ERROR,
+	INVALID_SUMBOLS,
+	DEFAULT_HOURS,
+} from 'constants/constants';
+
+import styles from './CreateCourse.module.css';
+
+import Button from 'common/Button/Button';
+import Input from 'common/Input/Input';
+import AuthorItem from './components/AuthorItem/AuthorItem';
+import { Form, Row, Col, FormGroup, List, Container } from 'reactstrap';
+
+import formatCreationDate from 'helpers/formatCreationDate.js';
+import formatDuration from 'helpers/formatDuration';
+
+import { Course, Author } from 'components/Courses/Course.types';
+
+type CourseFormProps = {
+	addCourse: Dispatch<SetStateAction<unknown>>;
+	addAuthor: Dispatch<SetStateAction<unknown>>;
+	setIsFormVisible: Dispatch<SetStateAction<unknown>>;
+	allAuthors: Author[];
+};
+
+const formInputs = {
+	id: uuidv4(),
+	title: '',
+	description: '',
+	creationDate: formatCreationDate(),
+	duration: '',
+	authors: [],
+};
+
+const CreateCourse = ({
+	addCourse,
+	addAuthor,
+	setIsFormVisible,
+	allAuthors,
+}: CourseFormProps) => {
+	const [authorName, setAuthorName] = useState('');
+	const [course, setCourse] = useState(formInputs);
+
+	const handleChange = (event) => {
+		const { name, value } = event.target;
+		if (!INVALID_SUMBOLS.test(value)) {
+			setCourse((prevState) => {
+				return {
+					...prevState,
+					[name]: value,
+				};
+			});
+		}
+	};
+
+	const handleAuthorDelete = (event) => {
+		const authorId = event.target.name;
+		setCourse((prevState) => {
+			return {
+				...prevState,
+				authors: prevState.authors.filter((author) => author.id !== authorId),
+			};
+		});
+	};
+
+	const handleAuthorCreate = () => {
+		if (!INVALID_SUMBOLS.test(authorName)) {
+			addAuthor((prevState) => [
+				...prevState,
+				{ id: uuidv4(), name: authorName },
+			]);
+		}
+		setAuthorName('');
+	};
+
+	const handleAuthorAdd = (event) => {
+		const authorId = event.target.name;
+		setCourse((prevState) => {
+			return {
+				...prevState,
+				authors: [
+					...prevState.authors,
+					allAuthors.find((author) => author.id === authorId),
+				],
+			};
+		});
+	};
+
+	const handleSubmit = (event) => {
+		if (
+			Object.values(course).every((input) => !!input) &&
```

Comment:
It looks like this line does nothing for you.
The checks below check all the fields you need.
![image](https://user-images.githubusercontent.com/41696704/226125879-8d35e2cf-0f15-4ec0-b00e-f98e0d77c994.png)